spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/ClientManagerBackToBackTests.java

```diff
@@ -247,6 +378,76 @@ public IntegrationFlow mqttInFlow(Mqttv5ClientManager mqttv5ClientManager) {
 
 	}
 
+	@Configuration
+	@EnableIntegration
+	public static class Mqttv5ConfigWithStartedManager {
+
+		static final String TOPIC_NAME = "test-topic-v5";
+
+		static final CountDownLatch subscribedLatch = new CountDownLatch(1);
+
+		@EventListener
+		public void onSubscribed(MqttSubscribedEvent e) {
+			subscribedLatch.countDown();
+		}
+
+		@Bean
+		public Mqttv5ClientManager mqttv5ClientManager() {
+			Mqttv5ClientManager manager = new Mqttv5ClientManager(MosquittoContainerTest.mqttUrl(), "client-manager-client-id-v5");
+			manager.start();
```

Comment:
You still didn't address this.
One more time: it is an anti-pattern to have component started too early.
Therefore the whole purpose of these `...WithStartedManager` configurations is wrong.
I suggestion to remove them altogether with their dedicated tests.
The regular (recommended) application context start is covered by other tests.
What we need for this change is really that based on the `IntegrationFlowContext` and beans added at runtime.
Therefore I'd expect only two new tests in your PR.

Thanks

Reply:
I am sorry.
Removed the tests and Messed up the history.  Thank you for your feedback and time. I think it must be hard to cooperate with someone like me. Hope everything is ok now.